indexer-app/src/cmr/indexer/services/index_service.clj

```diff
@@ -449,13 +458,27 @@
           ;; delete concept from primary concept index
           (do
             (es/delete-document
-             context index-names (concept-mapping-types concept-type)
-             concept-id revision-id elastic-version elastic-options)
-            ;; propagate collection deletion to granules and variables
+              context index-names (concept-mapping-types concept-type)
+              concept-id revision-id elastic-version elastic-options)
+            ;; Index a deleted-granule document when granule is deleted
+            (when (= :granule concept-type)
+              (let [es-doc (dg/deleted-granule->elastic-doc concept)]
+                (es/save-document-in-elastic
+                  context [deleted-granule-index-name] deleted-granule-type-name
+                  es-doc concept-id revision-id elastic-version elastic-options)))
+            ;; propagate collection deletion to granules
             (when (= :collection concept-type)
               (cascade-collection-delete
                context concept-mapping-types concept-id revision-id))))))))
 
+(defn remove-deleted-granule
```

Comment:
NIT: move this function into `cmr.indexer.data.concepts.deleted-granule` namespace.